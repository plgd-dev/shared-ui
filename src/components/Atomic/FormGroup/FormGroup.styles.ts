import { css } from '@emotion/react'
import { ThemeType, getThemeColor } from '../_theme'

export const formGroupMargin = css`
    margin-bottom: 20px;
`

export const inline = (inlineJustifyContent: string) => css`
    display: flex;
    align-items: center;
    justify-content: ${inlineJustifyContent};
`

export const errorMessage = (theme: ThemeType) => css`
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 160%;
    text-align: right;
    color: ${getThemeColor(theme, `FormGroup.error.color`)};
    padding-top: 4px;
`

export const inlineItems = css`
    margin: -4px;
    align-items: flex-start;
`

export const inlineItem = css`
    margin: 4px;
`

export const inlineItemFullSize = css`
    width: 100%;
`
