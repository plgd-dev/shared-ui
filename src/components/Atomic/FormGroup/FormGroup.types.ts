import { CSSProperties, ReactNode } from 'react'

export type Props = {
    children: ReactNode | ReactNode[]
    className?: string
    compactFormComponentsView?: boolean
    dataTestId?: string
    error?: string | boolean
    errorTooltip?: boolean
    fullSize?: boolean
    id: string
    inline?: boolean
    inlineJustifyContent?: string
    marginBottom?: boolean
    renderProps?: boolean
    style?: CSSProperties
    tooltipPortalTarget?: HTMLElement
    tooltipZIndex?: number
}

export const defaultProps = {
    marginBottom: true,
    inlineJustifyContent: 'flex-start',
}
