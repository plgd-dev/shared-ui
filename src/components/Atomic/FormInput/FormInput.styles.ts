import { css } from '@emotion/react'
import { ThemeType, getThemeColor, getTheme } from '../_theme'
import { fontPrimary } from '../_utils/commonStyles'

export const input = (theme: ThemeType, error?: boolean) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    border: 1px solid ${getThemeColor(theme, `FormInput.borderColor`)};
    background: ${getThemeColor(theme, `FormInput.background`)};
    border-radius: 8px;
    display: block;
    width: 100%;
    padding: 0 16px;
    transition: all 0.25s;
    color: ${getThemeColor(theme, `FormInput.color`)};
    box-sizing: border-box;

    &::placeholder {
        color: ${getThemeColor(theme, `FormInput.placeholder.color`)};
        opacity: 1;
    }

    &:-ms-input-placeholder {
        color: ${getThemeColor(theme, `FormInput.placeholder.color`)};
    }

    &::-ms-input-placeholder {
        color: ${getThemeColor(theme, `FormInput.placeholder.color`)};
    }

    &:focus {
        border-color: ${error ? undefined : getThemeColor(theme, `FormInput.focus.borderColor`)}!important;
        outline: none;
    }

    &::-ms-reveal,
    &::-ms-clear {
        display: none;
    }

    /* Chrome, Safari, Edge, Opera */
    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
    }

    /* Firefox */
    &[type='number'] {
        -moz-appearance: textfield;
    }
`

export const inlineStyle = (theme: ThemeType) => css`
    border-color: transparent;

    &:hover {
        border-color: ${getThemeColor(theme, `FormInput.borderColor`)};
    }
`

export const big = css`
    height: 48px;
`

export const normal = css`
    height: 44px;
`

export const small = css`
    height: 36px;
`

export const inputWithIconWrapper = css`
    position: relative;
    width: 100%;
`

export const inputWithIcon = css`
    padding-right: 48px;
`

export const inputTel = css`
    padding-left: 100px;
`

export const rightContent = css`
    position: absolute;
    right: 12px;
    top: 50%;
    transform: translateY(-50%);
    width: 24px;
    height: 24px;
    display: flex;
    align-items: center;
    justify-content: center;
    z-index: 2;
`

export const inputIcon = (theme: ThemeType) => css`
    color: ${getThemeColor(theme, `FormInput.icon.color`)};
    cursor: pointer;
    transition: all 0.25ms;

    &:hover {
        color: ${getThemeColor(theme, `FormInput.icon.hover.color`)};
    }

    svg {
        width: 24px;
        height: 24px;
    }
`

export const passwordIcon = (theme: ThemeType) => css`
    top: 10px;
    right: 12px;
    padding: 2px 0 2px 4px;
    background: ${getThemeColor(theme, `FormInput.passwordIcon.background`)};
    width: 28px;
    height: 28px;
    box-sizing: border-box;
`

export const inputTelData = (theme: ThemeType) => css`
    position: absolute;
    left: 12px;
    top: 12px;
    height: 24px;
    border-right: 1px solid ${getThemeColor(theme, `FormInput.borderColor`)};
    display: flex;
    align-items: center;
`

export const telPrefix = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    color: ${getThemeColor(theme, `FormInput.color`)};
    padding: 0 10px 0 8px;
    user-select: none;
`

export const disabled = (theme: ThemeType) => css`
    background: ${getThemeColor(theme, `FormInput.disabled.background`)};
    color: ${getThemeColor(theme, `FormInput.disabled.color`)};
    border-color: ${getThemeColor(theme, `FormInput.disabled.borderColor`)};
    cursor: not-allowed;
`

export const readOnly = (theme: ThemeType) => css`
    color: ${getThemeColor(theme, `FormInput.readonly.color`)};
`

export const error = (theme: ThemeType) => css`
    border-color: ${getThemeColor(theme, `FormInput.error.red`)}!important;
`

export const right = css`
    text-align: right;
    cursor: text;
`
