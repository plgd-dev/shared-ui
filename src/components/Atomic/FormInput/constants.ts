export const inputSizes = {
    BIG: 'big',
    NORMAL: 'normal',
    SMALL: 'small',
} as const

export const inputAligns = {
    LEFT: 'left',
    RIGHT: 'right',
} as const
