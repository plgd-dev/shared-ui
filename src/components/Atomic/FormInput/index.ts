export { default } from './FormInput'
export * from './FormInput'

export * from './constants'
