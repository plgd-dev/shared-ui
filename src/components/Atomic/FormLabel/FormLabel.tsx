import React, { forwardRef } from 'react'
import { Props } from './FormLabel.types'
import * as styles from './FormLabel.styles'
import ConditionalWrapper from '../ConditionalWrapper'
import { convertSize } from '../Icon'
import Tooltip from '../Tooltip'
import IconQuestion from '../Icon/components/IconQuestion'

const FormLabel = forwardRef<HTMLLabelElement, Props>((props, ref) => {
    const { className, id, inline, marginBottom, required, text, tooltipText, tooltipMaxWidth } = props
    return (
        <label
            className={className}
            css={[styles.label, inline && styles.inline, tooltipText && styles.w100, marginBottom === false && styles.noMargin]}
            htmlFor={id}
            ref={ref}
        >
            <ConditionalWrapper
                condition={!!tooltipText}
                wrapper={(c) => (
                    <span css={styles.flex}>
                        <span>{c}</span>
                        <Tooltip content={tooltipText} delay={200} maxWidth={tooltipMaxWidth} portalTarget={document.getElementById('modal-root')!}>
                            <IconQuestion {...convertSize(16)} />
                        </Tooltip>
                    </span>
                )}
            >
                <>
                    {text}
                    {required && <span css={styles.required}>*</span>}
                </>
            </ConditionalWrapper>
        </label>
    )
})

FormLabel.displayName = 'FormLabel'

export default FormLabel
