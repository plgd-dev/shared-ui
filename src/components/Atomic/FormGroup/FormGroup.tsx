import React, { Children, cloneElement, FC, isValidElement, ReactElement } from 'react'
import { defaultProps, Props } from './FormGroup.types'
import * as styles from './FormGroup.styles'
import Button from '../Button'

const FormGroup: FC<Props> = (props) => {
    const { children, className, error, id, inline, fullSize, marginBottom, inlineJustifyContent } = { ...defaultProps, ...props }
    const childrenWithProps = Children.map(children as ReactElement[], (child, key) => {
        if (isValidElement(child)) {
            if (child.type === Button) {
                return child
            }

            return cloneElement(child as ReactElement, { id, error: !!error, key, inline })
        }
        return child
    })

    const Error = error && typeof error === 'string' && (
        <div aria-live='polite' css={styles.errorMessage}>
            {error}
        </div>
    )

    const getInputs = () => {
        if (inline) {
            return (
                <>
                    <div css={styles.inline(inlineJustifyContent)}>{childrenWithProps}</div>
                    {Error}
                </>
            )
        } else if ((children as ReactElement[])?.length === 3) {
            return (
                <>
                    {childrenWithProps[0]}
                    <div css={[styles.inline(inlineJustifyContent), styles.inlineItems]}>
                        <div css={[styles.inlineItem, styles.inlineItemFullSize]}>
                            {childrenWithProps[1]}
                            {Error}
                        </div>
                        <div css={styles.inlineItem}>{childrenWithProps[2]}</div>
                    </div>
                </>
            )
        } else {
            return (
                <>
                    {childrenWithProps}
                    {Error}
                </>
            )
        }
    }

    return (
        <div className={className} css={[marginBottom && !inline && styles.formGroupMargin, fullSize && styles.inlineItemFullSize]}>
            {getInputs()}
        </div>
    )
}

FormGroup.displayName = 'FormGroup'
FormGroup.defaultProps = defaultProps

export default FormGroup
