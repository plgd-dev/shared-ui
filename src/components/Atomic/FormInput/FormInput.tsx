import { forwardRef, useContext, useRef, useState } from 'react'
import { detect } from 'detect-browser'
import { mergeRefs } from 'react-merge-refs'

import { Props, defaultProps } from './FormInput.types'
import * as styles from './FormInput.styles'
import { convertSize, IconCopy, IconHidePassword, IconShowPassword } from '../Icon'
import { copyToClipboard } from '../../../common/utils'
import { inputAligns, inputSizes } from './constants'
import { isEdge } from '../_utils/browser'
import { FormContext } from '../../../common/context/FormContext'

export const FormInputCore = forwardRef<HTMLInputElement, Props>((props, ref) => {
    const {
        align,
        ariaInvalid,
        autoComplete,
        copy,
        dataTestId,
        disabled,
        error,
        icon,
        inline,
        inlineStyle,
        inputWrapperStyle,
        inputRef,
        rightContent,
        size,
        readOnly,
        telPattern,
        telPrefix,
        type: defaultType,
        value,
        ...rest
    } = { ...defaultProps, ...props }
    const [type, setType] = useState(defaultType)
    const isEdgeBrowser = isEdge(detect())
    const localInputRef = useRef<HTMLInputElement>(null)
    const inputBase = (
        <input
            {...rest}
            aria-invalid={error || ariaInvalid ? true : undefined}
            autoComplete={autoComplete}
            css={(theme) => [
                styles.input(theme, error),
                (defaultType === 'password' || copy) && !isEdgeBrowser && styles.inputWithIcon,
                defaultType === 'tel' && styles.inputTel,
                disabled && styles.disabled(theme),
                error && styles.error,
                readOnly && styles.readOnly(theme),
                inlineStyle && styles.inlineStyle(theme),
                size === inputSizes.BIG && styles.big,
                size === inputSizes.NORMAL && styles.normal,
                size === inputSizes.SMALL && styles.small,
                align === inputAligns.RIGHT && styles.right,
            ]}
            data-endge-pass={isEdgeBrowser ? 'true' : undefined}
            data-inline={inline?.toString()}
            data-test-id={dataTestId}
            disabled={disabled || false}
            onFocus={
                align === inputAligns.RIGHT
                    ? () => {
                          setTimeout(() => {
                              localInputRef.current?.select()
                          }, 100)
                      }
                    : undefined
            }
            pattern={telPattern}
            readOnly={readOnly}
            ref={mergeRefs([ref, localInputRef, inputRef]) as any}
            type={type}
            value={value}
        />
    )

    if (defaultType === 'password' || defaultType === 'tel' || copy || icon || rightContent) {
        return (
            <div css={[styles.inputWithIconWrapper, inputWrapperStyle]}>
                {defaultType === 'tel' && (
                    <div css={styles.inputTelData} onClick={() => localInputRef?.current?.focus()}>
                        {icon}
                        <span css={styles.telPrefix}>{telPrefix}</span>
                    </div>
                )}
                {inputBase}
                {copy && (
                    <span css={[styles.inputIcon, styles.rightContent]} onClick={() => copyToClipboard(localInputRef.current?.value || '')}>
                        <IconCopy {...convertSize(24)} />
                    </span>
                )}
                {defaultType === 'password' && !copy && (
                    <span
                        css={[styles.inputIcon, isEdgeBrowser && defaultType === 'password' && styles.passwordIcon, styles.rightContent]}
                        onClick={() => setType(type === defaultType ? 'text' : defaultType)}
                    >
                        {type === 'text' ? <IconHidePassword {...convertSize(24)} /> : <IconShowPassword {...convertSize(24)} />}
                    </span>
                )}
                {defaultType !== 'password' && defaultType !== 'tel' && !copy && icon && <span css={[styles.inputIcon, styles.rightContent]}>{icon}</span>}
                {defaultType !== 'password' && defaultType !== 'tel' && !copy && !icon && <span css={styles.rightContent}>{rightContent}</span>}
            </div>
        )
    }

    return inputBase
})

const FormInput = forwardRef<HTMLInputElement, Props>((props, ref) => {
    const { commonInputProps, compactFormComponentsView } = useContext(FormContext)
    return <FormInputCore {...props} ref={ref} {...(compactFormComponentsView ? commonInputProps : {})} />
})

FormInput.displayName = 'FormInput'
FormInput.defaultProps = defaultProps

export default FormInput
