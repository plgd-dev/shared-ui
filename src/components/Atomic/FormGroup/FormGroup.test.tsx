import { render } from '../../../../test/jest-wrapper'
import FormGroup from './FormGroup'
import FormLabel from '../FormLabel'
import FormInput from '../FormInput'

describe('<FormGroup>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <FormGroup id='form-group-1'>
                    <FormLabel text='Placeholder text' />
                    <FormInput placeholder='Placeholder text' />
                </FormGroup>
                <FormGroup id='form-group-2'>
                    <FormLabel text='Placeholder text' />
                    <FormInput defaultValue='Default value text' />
                </FormGroup>
                <FormGroup id='form-group-3'>
                    <FormLabel text='Disabled text' />
                    <FormInput defaultValue='Disabled text' disabled={true} />
                </FormGroup>
                <FormGroup error='Error message' id='form-group-4'>
                    <FormLabel text='Error text' />
                    <FormInput defaultValue='Error text' />
                </FormGroup>
                <FormGroup id='form-group-5'>
                    <FormLabel text='Password' />
                    <FormInput defaultValue='Text' type='password' />
                </FormGroup>
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
