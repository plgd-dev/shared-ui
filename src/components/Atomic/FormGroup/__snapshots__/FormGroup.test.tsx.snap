// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 20px;
}

.emotion-1 {
  font-family: 'Poppins',sans-serif;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 160%;
  display: inline-block;
  margin-bottom: 10px;
  color: #81868C;
  cursor: pointer;
}

.emotion-2 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  border: 1px solid #D7D8DA;
  border-radius: 8px;
  display: block;
  width: 100%;
  padding: 0 16px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  color: #252626;
  box-sizing: border-box;
  height: 44px;
}

.emotion-2::-webkit-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-2::-moz-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-2:-ms-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-2::placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-2:-ms-input-placeholder {
  color: #81868C;
}

.emotion-2::-ms-input-placeholder {
  color: #81868C;
}

.emotion-2:focus {
  border-color: #2261AE!important;
  outline: none;
}

.emotion-2::-ms-reveal,
.emotion-2::-ms-clear {
  display: none;
}

.emotion-2::-webkit-outer-spin-button,
.emotion-2::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.emotion-2[type='number'] {
  -moz-appearance: textfield;
}

.emotion-8 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  border: 1px solid #D7D8DA;
  border-radius: 8px;
  display: block;
  width: 100%;
  padding: 0 16px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  color: #252626;
  box-sizing: border-box;
  background: #F6F7F9;
  color: #757676;
  border-color: #D7D8DA;
  cursor: not-allowed;
  height: 44px;
}

.emotion-8::-webkit-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-8::-moz-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-8:-ms-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-8::placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-8:-ms-input-placeholder {
  color: #81868C;
}

.emotion-8::-ms-input-placeholder {
  color: #81868C;
}

.emotion-8:focus {
  border-color: #2261AE!important;
  outline: none;
}

.emotion-8::-ms-reveal,
.emotion-8::-ms-clear {
  display: none;
}

.emotion-8::-webkit-outer-spin-button,
.emotion-8::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.emotion-8[type='number'] {
  -moz-appearance: textfield;
}

.emotion-11 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  border: 1px solid #D7D8DA;
  border-radius: 8px;
  display: block;
  width: 100%;
  padding: 0 16px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  color: #252626;
  box-sizing: border-box;
  border-color: #D74E3A!important;
  height: 44px;
}

.emotion-11::-webkit-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-11::-moz-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-11:-ms-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-11::placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-11:-ms-input-placeholder {
  color: #81868C;
}

.emotion-11::-ms-input-placeholder {
  color: #81868C;
}

.emotion-11:focus {
  border-color: !important;
  outline: none;
}

.emotion-11::-ms-reveal,
.emotion-11::-ms-clear {
  display: none;
}

.emotion-11::-webkit-outer-spin-button,
.emotion-11::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.emotion-11[type='number'] {
  -moz-appearance: textfield;
}

.emotion-12 {
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 160%;
  text-align: right;
  color: #D74E3A;
  padding-top: 4px;
}

.emotion-15 {
  position: relative;
  width: 100%;
}

.emotion-16 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  border: 1px solid #D7D8DA;
  border-radius: 8px;
  display: block;
  width: 100%;
  padding: 0 16px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  color: #252626;
  box-sizing: border-box;
  padding-right: 48px;
  height: 44px;
}

.emotion-16::-webkit-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-16::-moz-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-16:-ms-input-placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-16::placeholder {
  color: #81868C;
  opacity: 1;
}

.emotion-16:-ms-input-placeholder {
  color: #81868C;
}

.emotion-16::-ms-input-placeholder {
  color: #81868C;
}

.emotion-16:focus {
  border-color: #2261AE!important;
  outline: none;
}

.emotion-16::-ms-reveal,
.emotion-16::-ms-clear {
  display: none;
}

.emotion-16::-webkit-outer-spin-button,
.emotion-16::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.emotion-16[type='number'] {
  -moz-appearance: textfield;
}

.emotion-17 {
  color: #81868C;
  cursor: pointer;
  -webkit-transition: all 0.25ms;
  transition: all 0.25ms;
  position: absolute;
  right: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 24px;
  height: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  z-index: 2;
}

.emotion-17:hover {
  color: #007BBF;
}

.emotion-17 svg {
  width: 24px;
  height: 24px;
}

<div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        for="form-group-1"
      >
        Placeholder text
      </label>
      <input
        class="emotion-2"
        id="form-group-1"
        name=""
        placeholder="Placeholder text"
        tabindex="1"
        type="text"
        value=""
      />
    </div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        for="form-group-2"
      >
        Placeholder text
      </label>
      <input
        class="emotion-2"
        id="form-group-2"
        name=""
        tabindex="1"
        type="text"
        value="Default value text"
      />
    </div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        for="form-group-3"
      >
        Disabled text
      </label>
      <input
        class="emotion-8"
        disabled=""
        id="form-group-3"
        name=""
        tabindex="1"
        type="text"
        value="Disabled text"
      />
    </div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        for="form-group-4"
      >
        Error text
      </label>
      <input
        aria-invalid="true"
        class="emotion-11"
        id="form-group-4"
        name=""
        tabindex="1"
        type="text"
        value="Error text"
      />
      <div
        aria-live="polite"
        class="emotion-12"
      >
        Error message
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <label
        class="emotion-1"
        for="form-group-5"
      >
        Password
      </label>
      <div
        class="emotion-15"
      >
        <input
          class="emotion-16"
          id="form-group-5"
          name=""
          tabindex="1"
          type="password"
          value="Text"
        />
        <span
          class="emotion-17"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 25 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2.375c-3.367 0-6.063 1.581-8.025 3.463-1.95 1.865-3.254 4.1-3.87 5.634-.138.339-.138.717 0 1.056.616 1.534 1.92 3.769 3.87 5.634 1.962 1.882 4.658 3.463 8.025 3.463s6.062-1.581 8.025-3.463c1.95-1.87 3.254-4.1 3.875-5.633.137-.34.137-.718 0-1.057-.62-1.534-1.925-3.769-3.875-5.634C18.062 3.956 15.367 2.375 12 2.375M18 12c0 3.416-2.688 6.188-6 6.188-3.313 0-6-2.772-6-6.188s2.687-6.187 6-6.187c3.312 0 6 2.771 6 6.187m-6-2.75C12 10.767 10.804 12 9.333 12c-.479 0-.929-.129-1.316-.36-.009.12-.017.235-.017.36 0 2.277 1.792 4.125 4 4.125s4-1.848 4-4.125-1.792-4.125-4-4.125q-.175-.001-.35.017c.22.4.35.864.35 1.358"
              fill="currentcolor"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
